### PR TITLE
Remove custom docker client initialization logic

### PR DESCRIPTION
**Pull request by aanand**
Use `docker.utils.kwargs_from_env` instead.

### CODE REVIEW

File: compose/cli/docker_client.py [hunk 1]
```diff
@@ -9,27 +8,9 @@ def docker_client():
     Returns a docker-py client configured using environment variables
     according to the same logic as the official Docker client.
     """
-    cert_path = os.environ.get('DOCKER_CERT_PATH', '')
-    if cert_path == '':
-        cert_path = os.path.join(os.environ.get('HOME', ''), '.docker')
```
Comment (dnephin) [inline compose/cli/docker_client.py]: From what I can see `kwargs_from_env()` doesn't include this default for a cert path, is that correct?  Is this path no longer used?

Comment (aanand) [inline compose/cli/docker_client.py]: Well spotted. Looks like that logic does indeed still exist in Docker:

- [read DOCKER_CERT_PATH](https://github.com/docker/docker/blob/master/docker/flags.go#L28)
- [set it to cliconfig.ConfigDir() if empty](https://github.com/docker/docker/blob/master/docker/flags.go#L76)
- [cliconfig sets it to $HOME/.docker if empty](https://github.com/docker/docker/blob/master/cliconfig/config.go#L33)

So I think we should add it to docker-py first - this PR isn't urgent.

Comment (aanand) [inline compose/cli/docker_client.py]: https://github.com/docker/docker-py/pull/686

**Comment by thaJeztah** [comment]
@aanand I see https://github.com/docker/docker-py/pull/686 was merged

**Comment by aanand** [comment]
Yep - we can merge this as soon as the next docker-py release is out.

**Comment by mnowster** [comment]
Closing in favour of https://github.com/docker/compose/pull/2078

**Pull request closed without merging**